YoutubeExplode/Videos/Streams/StreamClient.cs

```diff
@@ -194,17 +195,22 @@ public StreamClient(HttpClient httpClient)
             var signatureScrambler = playerSource?.TryGetSignatureScrambler() ?? SignatureScrambler.Null;
 
             var playerResponseFromWatchPage = watchPage.TryGetPlayerResponse();
+
             if (playerResponseFromWatchPage is not null)
             {
+                if (playerResponseFromWatchPage.TryGetAgeRestricted())
+                    //if it's null, it will continue with the previous playerResponse
+                    playerResponseFromWatchPage = await _controller.GetPlayerResponseFromEndpoint(videoId) ?? playerResponseFromWatchPage;
+
                 var purchasePreviewVideoId = playerResponseFromWatchPage.TryGetPreviewVideoId();
-                if (!string.IsNullOrWhiteSpace(purchasePreviewVideoId))
+                bool isPurchaseRequired = playerResponseFromWatchPage.TryGetPurchaseRequired();
```

Comment:
Do we need `isPurchaseRequired` in addition to `purchasePreviewVideoId`? It seems if the latter is not null, it's fair to assume that the video is age-restricted.